### DISCUSSION TITLE
Database-agnostic column default values

### DISCUSSION THREAD

**Discussion by hy144328** [sqlalchemy/sqlalchemy#8143] [Ideas]
I am migrating a script to migrate from one database to another.
I use an event listener to genericize the column types of the source database:
https://docs.sqlalchemy.org/en/14/core/reflection.html#reflecting-with-database-agnostic-types .
Nevertheless, this does not genericize the default value of the affected column.

For example, I have a MySQL 8.0 source database.
The DDL of the source column looks like this:
```
  `entered` timestamp(6) NOT NULL DEFAULT CURRENT_TIMESTAMP(6),
```
The DDL after genericizing would look like this:
```
  `entered` DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP(6),
```
I receive the error message:
```
pymysql.err.OperationalError: (1067, "Invalid default value for 'entered'")
```

Ideas:
*   Would it be difficult to equip the `Column` class with an `as_generic()` method that returns a column where `type`, `default`, `server_default`, etc. are sensibly genericized, at least for the major SQL distributions?
*   For now, I am just dropping the default value in the target database because it does not hurt my use case.
    Would it be in line with the design philosophy to equip `schema` classes such as `Table` and `Column` with `drop_()` methods?
    This is particularly useful when the schema is not self-defined but inferred through reflection?

I am not sure if this problem exclusively pertains to timestamps/datetimes: https://github.com/sqlalchemy/sqlalchemy/discussions/7253 .
Nevertheless, wanted to use the opportunity to engage with the community.
Big fan, but only a humble user so far.

**Comment by zzzeek** [comment]
well the "default" value is present in the event-reflected column_dict as a string.   you dont have a Column object yet.   So you can also modify that default string to be something that matches what you want to do.   We dont parse that default so there's no place that SQLAlchemy would have anything built in for that, the default can be literally any SQL expression so that's on the user side to handle.

in this case however MySQL has a very specific thing going on with the above CURRENT_TIMESTAMP thing, is it just the precision of 6 that's the issue or is it that CURRENT_TIMESTAMP is not compatible with DATETIME ?   I have only vague recollection of the situation here although I do know it's a very specific case.
